specs/UpdateRuntime.md

```diff
@@ -0,0 +1,229 @@
+  /// information about the failure.
+  COREWEBVIEW2_UPDATE_RUNTIME_STATUS_FAILED,
+} COREWEBVIEW2_UPDATE_RUNTIME_STATUS;
+
+
+/// The UpdateRuntime operation result.
+[uuid(DD503E49-AB19-47C0-B2AD-6DDD09CC3E3A), object, pointer_default(unique)]
+interface ICoreWebView2ExperimentalUpdateRuntimeResult : IUnknown {
+
+  /// The status for the UpdateRuntime operation.
+  [propget] HRESULT Status(
+      [ out, retval ] COREWEBVIEW2_UPDATE_RUNTIME_STATUS * status);
+
+  /// The update error happened while trying to update Edge WebView2 Runtime.
+  [propget] HRESULT UpdateError([out, retval] HRESULT* updateError);
+}
+
+/// The caller implements this interface to receive the UpdateRuntime result.
+[uuid(F1D2D722-3721-499C-87F5-4C405260697A), object, pointer_default(unique)]
+interface ICoreWebView2ExperimentalUpdateRuntimeCompletedHandler : IUnknown {
+
+  /// Provides the result for the UpdateRuntime operation.
+  /// `errorCode` will be S_OK if the update operation can be performed
+  /// normally, regardless of whether we could update the Edge WebView2
+  /// Runtime. If an unexpected error interrupts the update operation, error
+  /// code of that unexpected error would be set as `errorCode`.
+  /// When update operation can be performed normally, but update resulted in
+  /// failure, like download failed, the error code would be presented as
+  /// `UpdateError` property of ICoreWebView2ExperimentalUpdateRuntimeResult.
+  HRESULT Invoke([in] HRESULT errorCode,
+                 [in] ICoreWebView2ExperimentalUpdateRuntimeResult * result);
+}
+
+/// This interface is an extension of the ICoreWebView2Environment. An object
+/// implementing the ICoreWebView2ExperimentalEnvironment3 interface will also
+/// implement ICoreWebView2Environment.
+[uuid(9A2BE885-7F0B-4B26-B6DD-C969BAA00BF1), object, pointer_default(unique)]
+interface ICoreWebView2ExperimentalEnvironment3 : IUnknown {
+  /// Try to update the installed Microsoft Edge WebView2 Runtime.
+  /// This will potentially result in a new version of the Edge WebView2
+  /// Runtime being installed and `NewBrowserVersionAvailable` event being raised.
+  /// There is no guarantee on the order of that event being raised and
+  /// UpdateRuntime's completed handler being invoked. Besides the
+  /// `NewBrowserVersionAvailable` event, there will be no impact to any
+  /// currently running WebView2s when the update is installed.
+  /// The latest version can always be queried using the
+  /// `GetAvailableCoreWebView2BrowserVersionString` API.
+  /// The UpdateRuntime method is only supported for an installed Edge WebView2
+  /// Runtime. When running a fixed version Edge WebView2 Runtime or non stable
+  /// channel Edge browser, this API will return `HRESULT_FROM_WIN32(ERROR_NOT_SUPPORTED)`.
+  /// There could only be one active UpdateRuntime operation, calling this API
+  /// before completed handler for previous call is invoked will fail with
+  /// `HRESULT_FROM_WIN32(ERROR_BUSY)`.
```

Comment:
Should also be a Status.

Reply:
If this applies to more than one app at a time then this is difficult for a particular app to control and we should rethink this restriction.

Reply:
This is per app process not global. Globally, it will be reported as failed update with error code of HRESULT_FROM_WIN32(ERROR_INSTALL_ALREADY_RUNNING). Will document that and leave this ERROR_BUSY as failure.

Reply:
Reading the feedback again, we should have a Status for install already running.

Reply:
That reintroduces the bug where multiple success status codes (one of which is common and the other is rare) leads to people failing to check the rare one.

Reply:
It is not a success for the case where there is already an install running globally. That running install could also fail. The new status just mean that the request from the app is rejected by the global edge update service and the app should retry to be sure that we are running latest version.
As we cannot report install already running as success and Edge update service doesn't support "joining" the operation, we would have to report this as a new status or failed status with some special error code. A new status code seems better to me.